app/scripts/directives/oscSourceSecrets.js

```diff
@@ -0,0 +1,93 @@
+"use strict";
+
+angular.module("openshiftConsole")
+
+  .directive("oscSourceSecrets", function($uibModal, $filter, DataService, SecretsService) {
+    return {
+      restrict: 'E',
+      scope: {
+        pickedSecrets: "=model",
+        secretsByType: '=',
+        strategyType: '=',
+        type: "@",
+        displayType: "@",
+        namespace: "=",
+        alerts: '=',
+        serviceAccountToLink: '@?'
+      },
+      templateUrl: 'views/directives/osc-source-secrets.html',
+      controller: function($scope) {
```

Comment:
Any reason we can't add these function to `link` instead of `controller` and remove `controller`?